tianshou/highlevel/experiment.py

```diff
@@ -372,19 +463,29 @@ def run(
 
 
 class ExperimentBuilder:
+    """A helper class (following the builder pattern) for creating experiments.
+
+    It contains a lot of defaults for the setup which can be adjusted using the
+    various `with_` methods. For example, the default optimizer is Adam, but can be
+    adjusted using :meth:`with_optim_factory`. Moreover, for simply configuring the default
+    optimizer instead of using a different one, one can use :meth:`with_optim_factory_default`.
+    """
+
     def __init__(
         self,
         env_factory: EnvFactory,
         experiment_config: ExperimentConfig | None = None,
         sampling_config: SamplingConfig | None = None,
     ):
-        if experiment_config is None:
-            experiment_config = ExperimentConfig()
-        if sampling_config is None:
-            sampling_config = SamplingConfig()
-        self._config = experiment_config
+        """:param env_factory: controls how environments are to be created.
+        :param experiment_config: the configuration for the experiment. If None, will use the default values
+            of `ExperimentConfig`.
+        :param sampling_config: the sampling configuration to use. If None, will use the default values
+            of `SamplingConfig`.
+        """
+        self._config = experiment_config or ExperimentConfig()
```

Comment:
Not a big fan of this pattern because it relies on implicit Boolean coercion. 
Why did you change my perfectly fine implementation, which was more explicit?

Reply:
I'll change back, it's just an aesthetic preference

Reply:
Done